crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -96,7 +96,7 @@ impl Settlement {
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
     #[error("failed communication with the database: {0}")]
-    Infra(sqlx::Error),
+    Infra(anyhow::Error),
```

Comment:
why did this change? Where are we turning a non sqlx error into a database error (since the to_string annotation still mentions databased communication failure)?

I think having an event index that doesn't fit into the right type could be considered `InconsistentData` (the existing error variant just below)

Reply:
`DatabaseError(sqlx::Error)` was changed to `DatabaseError(anyhow::Error)` since I wanted to get rid of the `error::Settlement` and `error::SettlementEvent` and convert `InvalidEvent` into `DatabaseError`, and this was not possible for `DatabaseError(sqlx::Error)`.

Converting `DatabaseError(sqlx::Error)` to `DatabaseError(anyhow::Error)` is probably fine since all db errors are treated the same at the callsite.